utils/interface.py

```diff
@@ -0,0 +1,256 @@
+
+    def __init__(self, baudrate, path, bonjour):
+        self.path = path
+
+        self.is_reading = False
+
+        self.header = ""
+        self.data = []
+        self.calibration = {}
+        self.messages = []
+
+        self.date_created = datetime.datetime.now().replace(microsecond=0).isoformat()
+
+        self.data_file = "{}_{}_data.csv".format(
+            self.date_created.replace(":", "-"),
+            bonjour.lower())
+        self.data_path = join(self.path, self.data_file)
+
+        self.calibration_file = "{}_{}_calibration.yaml".format(
+            self.date_created.replace(":", "-"),
+            bonjour.lower())
+        self.calibration_path = join(self.path, self.calibration_file)
+
+        self.messages_file = "{}_{}_messages.log".format(
+            self.date_created.replace(":", "-"),
+            bonjour.lower())
+        self.messages_path = join(self.path, self.messages_file)
+
+        if not isdir(self.path):
+            mkdir(self.path)
+
+        self.serial = SerialWrapper(baudrate=baudrate, bonjour=bonjour)
+
+    def write_data(self, dataArray):
+        """ Append a line in the file located at `self.data_path`
+
+        Each element of `dataArray` is written separated by a comma ','
+        If the file does not exit, it is created
+
+        Parameters
+        ----------
+        dataArray: array-like object
+            data to write in the file
+
+        """
+        with open(self.data_path, 'a+', newline='') as file:
+            writer = csv.writer(file, delimiter=',')
+            writer.writerow(dataArray)
+
+    def process_header(self, line):
+        """ Save header in memory
+
+        The first item is the computer time of reception
+
+        Parameters
+        ----------
+        line : string
+            line received from telemetry, without start and end character
+
+        """
+        # Don't write the header twice
+        if not self.header:
+            self.header = ["Computer Time"] + line.split(self.separators['SEP_DATA'])
+            self.write_data(self.header)
+            print("Header : {}".format(self.header))
+
+    def process_data(self, now, line):
+        """ Save data in memory and on file storage
+
+        The first item is the computer time of reception as a datetime.datetime object
+        Data cannot be recorded if the header has not been received
+
+        Parameters
+        ----------
+        now : datetime object
+            time of reception of the line
+        line : string
+            line received from telemetry, without start and end character
+
+        """
+        if self.header:
+            data = [now] + line.split(self.separators['SEP_DATA'])
+            self.data.append(data)
+            self.write_data(data)
+
+    def write_calibration(self):
+        """ Write the calibration data in the file located at `self.calibration_path`
+
+        """
+        with open(self.calibration_path, 'w') as file:
```

Comment:
This is a new file for each calibration, right? Since the path is changed regarding time, if i understand it correctly?

Reply:
Yes. Each time we run the dashboard three new files are created to store the data, the messages and the calibration. This way no data is lost if we run the dashboard multiple times